homeassistant/components/simplepush/notify.py

```diff
@@ -61,11 +61,32 @@ def send_message(self, message: str, **kwargs: Any) -> None:
         """Send a message to a Simplepush user."""
         title = kwargs.get(ATTR_TITLE, ATTR_TITLE_DEFAULT)
 
+        attachments = None
         # event can now be passed in the service data
         event = None
         if data := kwargs.get(ATTR_DATA):
             event = data.get(ATTR_EVENT)
 
+            attachments_data = data.get(ATTR_ATTACHMENTS)
+            if isinstance(attachments_data, list) and attachments_data:
+                attachments = []
+                for attachment in attachments_data:
+                    if (
+                        isinstance(attachment, dict)
+                        and "video" in attachment.keys()
+                        and "thumbnail" in attachment.keys()
+                    ):
+                        attachments.append(
+                            {
+                                "video": attachment["video"],
+                                "thumbnail": attachment["thumbnail"],
+                            }
+                        )
+                    elif isinstance(attachment, dict) and "video" in attachment.keys():
+                        attachments.append(attachment["video"])
+                    else:
+                        attachments.append(attachment)
+
```

Comment:
A better approach would be to raise an error if the `attachments` key values are not as per the documentation. If no errors are raised then simply use `attachments=attachments_data`.
for lines 85 and 86 I believe a `video` key should only be provided with a thumbnail. The documentation should mentions this clearly so no need to check for a video only key in the attachments.

Reply:
Thanks, makes a lot of sense.

Please see my latest commit for the implementation of your comment.